mcp/common/src/main/java/org/springframework/ai/mcp/ToolHelper.java

```diff
@@ -0,0 +1,184 @@
+/*
+* Copyright 2025 - 2025 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.springframework.ai.mcp;
+
+import java.util.List;
+
+import io.modelcontextprotocol.server.McpServerFeatures;
+import io.modelcontextprotocol.server.McpServerFeatures.AsyncToolRegistration;
+import io.modelcontextprotocol.spec.McpSchema;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import org.springframework.ai.model.ModelOptionsUtils;
+import org.springframework.ai.tool.ToolCallback;
+
+/**
+ * Utility class that provides helper methods for working with Model Context Protocol
+ * (MCP) tools in a Spring AI environment. This class facilitates the integration between
+ * Spring AI's tool callbacks and MCP's tool system.
+ *
+ * <p>
+ * The MCP tool system enables servers to expose executable functionality to language
+ * models, allowing them to interact with external systems, perform computations, and take
+ * actions in the real world. Each tool is uniquely identified by a name and includes
+ * metadata describing its schema.
+ *
+ * <p>
+ * This helper class provides methods to:
+ * <ul>
+ * <li>Convert Spring AI's {@link ToolCallback} instances to MCP tool registrations</li>
+ * <li>Generate JSON schemas for tool input validation</li>
+ * </ul>
+ *
+ * @author Christian Tzolov
+ */
+public final class ToolHelper {
```

Comment:
`ToolUtils` instead?  We have adopted the `Utils` suffix over time.

Reply:
sure